ephios/plugins/federation/views/api.py

```diff
@@ -0,0 +1,125 @@
+from urllib.parse import urljoin
+
+import requests
+from django.conf import settings
+from django.shortcuts import get_object_or_404, redirect
+from django.urls import reverse
+from django.views import View
+from oauth2_provider.contrib.rest_framework import TokenHasScope
+from oauthlib.oauth2 import WebApplicationClient
+from requests_oauthlib import OAuth2Session
+from rest_framework.exceptions import PermissionDenied
+from rest_framework.generics import CreateAPIView, ListAPIView
+from rest_framework.permissions import AllowAny
+
+from ephios.core.models import Event, Qualification
+from ephios.plugins.federation.models import FederatedGuest, FederatedUser
+from ephios.plugins.federation.serializer import (
+    FederatedGuestCreateSerializer,
+    SharedEventSerializer,
+)
+
+
+class RedeemInviteCodeView(CreateAPIView):
+    serializer_class = FederatedGuestCreateSerializer
+    queryset = FederatedGuest.objects.all()
+    authentication_classes = []
+    permission_classes = [AllowAny]
+
+
+class SharedEventListView(ListAPIView):
+    serializer_class = SharedEventSerializer
+    permission_classes = [TokenHasScope]
+    required_scopes = []
+
+    def get_queryset(self):
+        try:
+            guest = self.request.auth.federatedguest_set.get()
```

Comment:
I was wondering why this doesn't break if I create multiple connections between the same instances, as FederatedGuest has a ForeignKey, not a OneToOne key. But of course access token differ between FederatedGuests. Still could be OneToOne couldn't it?